tsk/fs/btrfs.cpp

```diff
@@ -1262,7 +1262,8 @@ static bool
 btrfs_treenode_push(BTRFS_INFO * a_btrfs, BTRFS_TREENODE ** a_node,
     const TSK_DADDR_T a_address, const BTRFS_DIRECTION a_initial_index)
 {
-    uint8_t raw[a_btrfs->sb->nodesize];
+    // was: uint8_t raw[a_btrfs->sb->nodesize];
+    uint8_t *raw = new uint8_t[a_btrfs->sb->nodesize];
```

Comment:
Put it in a `std::unique_ptr`. Then you don't need the `delete`s and you can't leak it.

Reply:
Do this:

```
std::unique_ptr<uint8_t[]> raw{new uint8_t[a_btrfs->sb->nodesize]};
```

After that, you need to access the pointer using `raw.get()`, but otherwise you should need no other changes. The allocation will be cleaned up when `std::unique_ptr` goes out of scope.

We should eventually replace every `malloc` and bare `new` for with `std::unique_ptr` (or `std::string`, if it's a string) for dynamic memory that's scope constrained.